### ISSUE TITLE
No longer seems to work

### ISSUE THREAD

**Issue by kergoth** [gdiepen/volume-sharer#3]
Starting the VM with ` docker run --name volume-sharer --rm -v /var/lib/docker/volumes:/docker_volumes -p 139:139 -p 445:445 -v /var/run/docker.sock:/var/run/docker.sock --net=host -d gdiepen/volume-sharer` Results in `WARNING: Published ports are discarded when using host network mode` and I'm unable to connect to the docker vm via smb in explorer, so it seems something has changed with how host networking works in current Docker for Windows, unless I'm missing something. Thoughts?

**Comment by gdiepen** [comment]
That is weird....
I tried to quickly find this warning message in the source code of docker (https://github.com/docker/docker-ce/blame/master/components/engine/daemon/container.go) and did see it was added 6 months ago.

Which version of windows are you using?

Unfortunately, I don't have a windows 10 computer at the moment where I can test this on, but it appears that this indeed has changed.

Can you check what happens in the case you leave out the -p 139:139 and -p 445:445?

After starting, can you do a portscan on the IP address of the docker machine?

**Comment by kergoth** [comment]
Yes, this is Windows 10 with the spring update applied. After disabling both my VPN and firewall just to make sure nothing was blocking it, I verified that docker's drive sharing was disabled too, since it uses smb under the hood, and verified that nothing was listening on 139 or 445 on 10.0.75.2.

I then fired up this container, despite the warnings about the publish, it's now listening on 139 and 445, but attempts to connect to it still fail, which is interesting:
```
PS C:\users\kergoth\apps\portqry> net view 10.0.75.2
System error 53 has occurred.

The network path was not found.

PS C:\users\kergoth\apps\portqry> .\PortQry.exe -n 10.0.75.2 -nr -o 139,445

Querying target system called:

 10.0.75.2

TCP port 139 (netbios-ssn service): LISTENING

TCP port 445 (microsoft-ds service): LISTENING
```

**Comment by gdiepen** [comment]
When you do a telnet to those ports, do you see a connection being made? Or is that connection refused.

Other things you might try is to take a look at the mobylinux vm (using something nsenter to get a shell into the virtual machine) and see if you see anything weird in there.

The docker drive sharing should not be an issue as that goes the other way around (i.e. the docker vm will connect to your windows machine using SMB)

Starting to run out of options ;)

**Comment by kergoth** [comment]
A connection is made just fine with telnet, dropping into the vm via nsenter shows it's listening on both ports in `netstat -nl` and `smbd` is running. Nothing obvious. Setting this aside for now, no time to spend on this, will just do without sharing the volumes for the time being. Thanks for your help thus far! FYI, it does show listening on both ports even without the publish arguments, so it seems it wasn't lying about ignoring it, but the ports show up anyway, though they still don't seem to actually work.

I also attempted to share one specific volume using someone elses samba docker image and it failed there as well, so this seems to be a semi-recent docker host networking change that prevents connecting to the vm via smb for some reason, which is affecting more images than just yours. Perhaps there's a workaround, but I don't yet know of one. Thanks again.

**Comment by dutchminator** [comment]
Still seems to work on our end, but I do get the same warning when starting the volume-sharer container. Very likely that your latest Windows updates have indeed caused this, instead of the docker update...

@kergoth Have you tried resetting Docker-for-Windows to factory default settings? Have you tried disabling and re-enabling the Hyper-V and container functionality in Windows? Recent Windows updates (With regards to LCOW, and using LinuxKit instead of MobyLinux) might have messed up some settings.

![2018-06-14 15_25_53-sebastestvolume](https://user-images.githubusercontent.com/9079330/41414978-bdba3f34-6fe7-11e8-9a0d-3b9acd105a62.png)

**Windows version:**
Windows 10 Enterprise, 1703, Build 15063.1088

**Docker version:**
 Client:
 Version:      18.03.1-ce
 API version:  1.37
 Go version:   go1.9.5
 Git commit:   9ee9f40
 Built:        Thu Apr 26 07:12:48 2018
 OS/Arch:      windows/amd64
 Experimental: false
 Orchestrator: swarm

Server:
 Engine:
  Version:      18.03.1-ce
  API version:  1.37 (minimum version 1.12)
  Go version:   go1.9.5
  Git commit:   9ee9f40
  Built:        Thu Apr 26 07:22:38 2018
  OS/Arch:      linux/amd64
  Experimental: true

**Comment by 0xR** [comment]
The same error message shows up. I am able to access the ipaddress via explorer, but I don't see any network shares: 
![image](https://user-images.githubusercontent.com/6420016/70990325-95144080-20c5-11ea-8742-94738dfc06c3.png)

```
PS C:\Users\USER> docker run --name volume-sharer --rm -v /var/lib/docker/volumes:/docker_volumes -p 139:139 -p 445:445 -v /var/run/docker.sock:/var/run/docker.sock --net=host -d gdiepen/volume-sharer
WARNING: Published ports are discarded when using host network mode
f70a198405f9bb08693e7a4841dc09b08a33683b368a0cc74d1b04df02f5a280
PS C:\Users\M64G555> docker ps
CONTAINER ID        IMAGE                   COMMAND              CREATED             STATUS              PORTS               NAMES
f70a198405f9        gdiepen/volume-sharer   "volume-sharer.sh"   3 seconds ago       Up 3 seconds                            volume-sharer
ee91a0021fa4        alpine                  "sh"                 3 minutes ago       Up 3 minutes                            priceless_poitras
```

The alpine image has a volume mounted: `docker run -v myvolume:/myvolume -it alpine sh`. I added a file using `echo asdf > /myvolume/file.txt`.

**Comment by Hamsterman** [comment]
I tried this:
docker run --name volume-sharer --rm -d -v /var/lib/docker/volumes:/docker_volumes -p 139:139 -p 445:445 -v /var/run/docker.sock:/var/run/docker.sock --net=host gdiepen/volume-sharer

C:\Program Files\Docker\docker.exe: Error response from daemon: Ports are not available: listen tcp 0.0.0.0:445: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted.

Then I tried this:
docker run --name volume-sharer --rm -d -v /var/lib/docker/volumes:/docker_volumes -v /var/run/docker.sock:/var/run/docker.sock --net=host gdiepen/volume-sharer

No problem starting the container. But it was not given any IP and no shares can be seen.
